a10_octavia/controller/worker/tasks/vthunder_tasks.py

```diff
@@ -345,21 +345,28 @@ class HandleACOSPartitionChange(task.Task):
 
     """Task to switch to specified partition"""
 
-    @axapi_client_decorator
     def execute(self, vthunder):
         try:
-            self.axapi_client.system.partition.create(vthunder.partition_name)
+            axapi_client = self.get_axapi_client(vthunder)
+            axapi_client.system.partition.create(vthunder.partition_name)
             LOG.info("Partition %s created", vthunder.partition_name)
         except acos_errors.Exists:
             pass
         except Exception as e:
             LOG.exception("Failed to create parition on vThunder: %s", str(e))
             raise
 
-    @axapi_client_decorator
     def revert(self, vthunder, *args, **kwargs):
         try:
-            self.axapi_client.system.partition.delete(vthunder.partition_name)
+            axapi_client = self.get_axapi_client(vthunder)
+            axapi_client.system.partition.delete(vthunder.partition_name)
         except Exception as e:
             LOG.exception("Failed to revert partition create : %s", str(e))
             raise
+
+    def get_axapi_client(self, vthunder):
```

Comment:
Let's move this to the util.py module. This way we can just make changes there instead of in decorators and here